src/HotChocolate/AspNetCore/src/AspNetCore/Types/FileType.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using HotChocolate.Language;
+using Microsoft.AspNetCore.Http;
+
+namespace HotChocolate.Types
+{
+    // todo: This is not fully implemented
+    public class FileValueNode : IValueNode<IFormFile>
```

Comment:
A value node represents a value literal. In this case we do not need a literal.

Reply:
I added it since I needed to return something valid for `ParseResult` of my `FileType` scalar. This made the most sense for me, with my limited understanding of the library. :) 
But as you pointed out in another comment, maybe the scalar approach is not the right one. I just copied what the [reference implementation](https://github.com/jaydenseric/graphql-upload/blob/master/public/GraphQLUpload.js) did and was satisfied when it worked out!

Reply:
Let me check what they are doing .... To be honest ... the whole spec is a bit wonky haha

I would have loved to have a more complete spec that also handles how files can be queried. I think we will do something in this direction.

Reply:
```js
module.exports = new GraphQLScalarType({
  name: 'Upload',
  description: 'The `Upload` scalar type represents a file upload.',
  parseValue(value) {
    if (value instanceof Upload) return value.promise;
    throw new GraphQLError('Upload value invalid.');
  },
  parseLiteral(ast) {
    throw new GraphQLError('Upload literal unsupported.', ast);
  },
  serialize() {
    throw new GraphQLError('Upload serialization unsupported.');
  },
});
```

So, the ref implementation just throws errors in this case.

I will have a look at how we tackle this problem.